### PR TITLE
Finish GraphQL setup and refactor

### PR DESCRIPTION
**Pull request by ARtoriouSs**

### CODE REVIEW

File: lib/squadster_web/schema/account_types.ex [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  use Absinthe.Schema.Notation
+
+  alias Squadster.Helpers.Dates
+
+  object :user do
+    field :id, non_null(:id)
+    field :uid, non_null(:string)
+    field :first_name, non_null(:string)
+    field :last_name, non_null(:string)
+    field :birth_date, :date
```
Comment (mirterious) [inline lib/squadster_web/schema/account_types.ex:11]: Is it necessary?

File: lib/squadster/accounts/user.ex [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    field :auth_token, :string
+    field :small_image_url, :string
+    field :image_url, :string
+    field :vk_url, :string
+    timestamps()
+  end
+
+  def changeset(params) do
+     changeset(%__MODULE__{}, params)
+  end
```
Comment (mirterious) [inline lib/squadster/accounts/user.ex:43]: whitespace??

**Review by mirterious — DISMISSED** [review]
I do not understand the logic of our app) (will watch through it at weekends)
So, there are some little cavils :)
Merge it by yourself, pls

File: lib/squadster_web/schema/account_types.ex (hunk 1 above)
Comment (ARtoriouSs) [reply lib/squadster_web/schema/account_types.ex:11]: why not? ¯\_(ツ)_/¯

File: lib/squadster/accounts/user.ex (hunk 2 above)
Comment (ARtoriouSs) [reply lib/squadster/accounts/user.ex:43]: fixed

File: lib/squadster/accounts.ex [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    |> put_change(:auth_token, nil)
+    |> Repo.update
+  end
+
+  def list_users do
+    Repo.all(User)
+  end
+
+  def find_user(id) do
+    Repo.get_by(User, id: id)
```
Comment (mirterious) [inline lib/squadster/accounts.ex]: maybe more clear
`User |> Repo.get_by(id: id)`
same in 2 methods below

File: lib/squadster/accounts/user.ex [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+      vk_url: @vk_url <> (info["domain"] || ""),
+      auth_token: token
+    }
+  end
+
+  def uid_from_auth(%Auth{extra: %{raw_info: %{user: %{"id" => uid}}}}) do
+    Integer.to_string(uid)
+  end
+
+  defp auth_fields do
```
Comment (mirterious) [inline lib/squadster/accounts/user.ex]: maybe make sence to define it as module variables
cos it doesn't provide any logic, just return an array

File: lib/squadster/helpers/dates.ex [hunk 5]
```diff
@@ -0,0 +1,19 @@
+defmodule Squadster.Helpers.Dates do
+  def date_from_string(string_date) do
+    [day, month, year] = string_date \
```
Comment (mirterious) [inline lib/squadster/helpers/dates.ex]: backslash is useless
image
image

File: lib/squadster_web/controllers/auth_controller.ex [hunk 6]
```diff
@@ -20,9 +20,8 @@ defmodule SquadsterWeb.SessionController do
   end
 
   def callback(%{assigns: %{ueberauth_auth: auth}} = conn, _params) do
-    case User.find_or_create(auth) do
-      {:ok, user} ->
-        redirect(conn, external: redirect_url(token: user.auth_token))
+    case Accounts.find_or_create_user(auth) do
+      {:ok, user}      -> redirect(conn, external: redirect_url(token: user.auth_token))
```
Comment (mirterious) [inline lib/squadster_web/controllers/auth_controller.ex]: this whitespace looks like a mess
maybe discard this changes
take a decision

File: lib/squadster_web/schema/account_types.ex (hunk 1 above)
Comment (mirterious) [reply lib/squadster_web/schema/account_types.ex:11]: yep, otherwise we do not need a module `Dates` :)

**Review by mirterious — CHANGES_REQUESTED** [review]

File: lib/squadster/helpers/dates.ex (hunk 5 above)
Comment (ARtoriouSs) [reply lib/squadster/helpers/dates.ex]: 👍

File: lib/squadster_web/controllers/auth_controller.ex (hunk 6 above)
Comment (ARtoriouSs) [reply lib/squadster_web/controllers/auth_controller.ex]: ok, why not

File: lib/squadster/accounts/user.ex (hunk 4 above)
Comment (ARtoriouSs) [reply lib/squadster/accounts/user.ex]: maybe
done

File: lib/squadster/accounts.ex (hunk 3 above)
Comment (ARtoriouSs) [reply lib/squadster/accounts.ex]: Fixed

**Pull request merged**